extension/json/src/functions/cast_functions/json_cast.cpp

```diff
@@ -0,0 +1,60 @@
+#include "common/exception/conversion.h"
+#include "common/json_common.h"
+#include "common/string_format.h"
+#include "function/scalar_function.h"
+#include "json_cast_functions.h"
+#include "json_creation_functions.h"
+#include "json_type.h"
+#include "json_utils.h"
+
+namespace kuzu {
+namespace json_extension {
+
+using namespace function;
+using namespace common;
+
+static void execFunc(const std::vector<std::shared_ptr<common::ValueVector>>& parameters,
+    const std::vector<common::SelectionVector*>& /*parameterSelVectors*/,
+    common::ValueVector& result, common::SelectionVector* resultSelVector, void* /*dataPtr*/) {
+    result.resetAuxiliaryBuffer();
+    for (auto i = 0u; i < resultSelVector->getSelSize(); ++i) {
+        auto resultPos = (*resultSelVector)[i];
+        auto str = parameters[0]->getValue<common::ku_string_t>(resultPos);
+        yyjson_read_err error;
+        auto doc = JSONCommon::readDocumentUnsafe(str.getDataUnsafe(), str.len,
+            JSONCommon::READ_FLAG, &error);
+        if (!doc) {
+            yyjson_doc_free(doc);
+            throw common::ConversionException{
+                common::stringFormat("Cannot convert {} to JSON.", str.getAsString())};
+        }
+        yyjson_doc_free(doc);
+        StringVector::addString(&result, resultPos, str);
+    }
+}
+
+static std::unique_ptr<FunctionBindData> bindFunc(ScalarBindFuncInput input) {
+    LogicalType type = input.arguments[0]->getDataType().copy();
+    if (type.getLogicalTypeID() == LogicalTypeID::ANY) {
+        type = LogicalType::INT64();
```

Comment:
When will this be triggered? I would have a test case on this.

Reply:
We already have a test for it. e.g. `cast(null as json)`